things/postgres/database.go

```diff
@@ -0,0 +1,60 @@
+package postgres
+
+import (
+	"context"
+	"database/sql"
+
+	"github.com/jmoiron/sqlx"
+	"github.com/opentracing/opentracing-go"
+)
+
+var _ DatabaseMiddleware = (*databaseMiddleware)(nil)
+
+type databaseMiddleware struct {
+	db *sqlx.DB
+}
+
+// DatabaseMiddleware provides a database interface
+type DatabaseMiddleware interface {
```

Comment:
Please rename this interface to something like `Database`, or `DB`. Don't want to be aware that this is middleware on an interface level.